apps/antalmanac/src/stores/calendarizeHelpers.ts

```diff
@@ -67,46 +58,46 @@ export const calendarizeFinals = (currentCourses: ScheduleCourse[] = []) => {
 
     for (const course of currentCourses) {
```

Comment:
```suggestion
const WEEK_DAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

/**
 * Given a string or string array of weekdays, return an array of booleans indicating whether
 * each weekday occurs in the input.
 */
function getWeekdaysOcurring(weekDays?: string | string[] | null): boolean[] {
    return weekDays ? WEEK_DAYS.map((day) => weekDays.includes(day)) : WEEK_DAYS.map(() => false);
}

export function calendarizeFinals(currentCourses: ScheduleCourse[] = []): CourseEvent[] {
    return currentCourses
        .filter((course) => course.section.finalExam.examStatus === 'SCHEDULED_FINAL')
        .filter(
            (course) =>
                course.section.finalExam.startTime &&
                course.section.finalExam.endTime &&
                course.section.finalExam.dayOfWeek
        )
        .flatMap((course) => {
            const finalExam = course.section.finalExam;
            const startHour = finalExam.startTime?.hour;
            const startMin = finalExam.startTime?.minute;
            const endHour = finalExam.endTime?.hour;
            const endMin = finalExam.endTime?.minute;

            const weekdaysOccurring = getWeekdaysOcurring(course.section.finalExam.dayOfWeek);
            return weekdaysOccurring.filter(Boolean).map((_day, index) => {
                return {
                    color: course.section.color,
                    term: course.term,
                    title: course.deptCode + ' ' + course.courseNumber,
                    courseTitle: course.courseTitle,
                    bldg: course.section.meetings[0].bldg[0],
                    instructors: course.section.instructors,
                    sectionCode: course.section.sectionCode,
                    sectionType: 'Fin',
                    start: new Date(2018, 0, index - 1, startHour, startMin),
                    end: new Date(2018, 0, index - 1, endHour, endMin),
                    finalExam: course.section.finalExam,
                    isCustomEvent: false,
                };
            });
        });
}
```
This function should be implemented like this

Reply:
This makes sense to me, but there were some minor kinks in the implementation here: 
`return weekdaysOccurring.filter(Boolean).map((_day, index) => {`

I think this change, as well as the similar one for `calendarizeFinals` would be best done in a separate PR where `calendarizeCourseEvents` and `CalendarizeFinals` are made a little less redundant.

Reply:
Wait, what was the issue?